src/conn/conn_prefetch.c

```diff
@@ -204,6 +207,14 @@ __wt_conn_prefetch_clear_tree(WT_SESSION_IMPL *session, bool all)
 
     __wt_spin_unlock(session, &conn->prefetch_lock);
 
+    /*
+     * To avoid a race condition, check if this dhandle is currently being operated on by any other
+     * prefetch thread. If so, wait for the thread to finish before proceeding with eviction.
+     */
+    if (dhandle != NULL)
+        while (((WT_BTREE *)dhandle->handle)->prefetch_busy > 0)
+            __wt_yield();
```

Comment:
I need some clarification here, so this is a single dhandle, how many threads can be incrementing prefetch_busy on this dhandle? Can the value be >1?.

If we see this value as 1, then sure we can wait for it be 0. But what's to stop it from appearing to be 0 while someone currently increments it to 1? So this thread doesn't "wait" but the value is still 1. Is that safe? If you're sychronizing these threads on this variable then either you block the other thread or you don't.